lib/pharos/cluster_manager.rb

```diff
@@ -22,15 +29,21 @@ def ssh_manager
     # @return [Pharos::AddonManager]
     def phase_manager
       @phase_manager = Pharos::PhaseManager.new(
+        @config,
         ssh_manager: ssh_manager,
+        kube_session: kube_session,
         config: @config,
         cluster_context: @context
       )
     end
 
     # @return [Pharos::AddonManager]
     def addon_manager
-      @addon_manager ||= Pharos::AddonManager.new(@config, @context)
+      @addon_manager ||= Pharos::AddonManager.new(
+        @config,
+        cluster_context: @context,
+        kube_session: kube_session
```

Comment:
> Need to delay the Pharos::Kube::Session creation to until it's actually needed by the PhaseManager / AddonManager... I suppose maybe the answer here could be to already introduce the Pharos::Kube::Manager that could later also be used for managing the per-thread sessions.

Didn't go for the `Pharos::Kube::Manager` yet, that would be over-engineering at this point... kept the non-threadsafe `Pharos::Kube#session` instead, it's simplest, and easy to continue improving that later because it's only used by the `PhaseManager`/`AddonManager` now.